src/extension.ts

```diff
@@ -1145,9 +1198,13 @@ function execKubernetes() {
 
 async function terminalKubernetes(explorerNode?: explorer.ResourceNode) {
     if (explorerNode) {
-        // For those images (e.g. built from Busybox) where bash may not be installed by default, use sh instead.
-        const isBash = await isBashOnPod(explorerNode.id);
-        execTerminalOnPod(explorerNode.id, isBash ? 'bash' : 'sh');
+        const podMetadata = { name: explorerNode.id, namespace: undefined };  // TODO: namespaces
+        const container = await selectContainerForPod(podMetadata);
+        if (container) {
+            // For those images (e.g. built from Busybox) where bash may not be installed by default, use sh instead.
+            const isBash = await isBashOnContainer(podMetadata.name, podMetadata.namespace, container.name);
```

Comment:
nit: maybe we can make isBashOnContainer return the shell to run on the container, since we have some pretty regular usage of it.